sycl/doc/ControllingSYCLLanguageFeatures.md

```diff
@@ -0,0 +1,195 @@
+# Controlling SYCL/DPC++ language features
+
+This documents aims to describe a mechanism which is used to control (i.e.
+enable or disable) different SYCL/DPC++ language and library features, like
+version of SYCL/DPC++ standard to use, set of extensions to use, etc.
+
+## Controlling SYCL/DPC++ standard version
+
+In order to control the language standard to compile for, the following options
+can be used:
+
+`-sycl-std=<value>`, `--sycl-std=<value>`, `--sycl-std <value>`
+
+SYCL language standard to compile for.
+
+Possible values:
+
+- `2017`, '121', `1.2.1`, `sycl-1.2.1`: corresponds to SYCL 1.2.1
+  specification, see [SYCL Registry] for more details and the specification
+  text.
+
+  Note: setting `-sycl-std` option automatically implies a particular C++
+  standard version to be set.
+
+  **TODO**: do we need to specify which exact version is implied by which
+  `-sycl-std` value? AFAIK, SYCL spec only specifies minimum required C++
+  version, which is C++11 and we set C++17 in our implementation
+
+- `dpcpp-0.7`: corresponds to DPC++ 0.7.
+
+  Basically, `-sycl-std=dpcpp-0.7` implies support for SYCL 1.2.1 specification,
+  bunch of extension enabled and `-std=c++17`
+
+  See [oneAPI Specification] for more details and the specification text.
+
+[SYCL Registry]: https://www.khronos.org/registry/SYCL/
+[oneAPI Specification]: https://spec.oneapi.com/
+
+Note: it is possible to change C++ version independently of SYCL/DPC++ standard
+version if that is needed: `-sycl-std=1.2.1 -std=c++14`, for example.
```

Comment:
```cpp
then it is expected to see compilation errors. 
```

Its unclear whether this is just going to be some random C++ compilation error because using the wrong interface (e.g., lack of CTAD would cause strange template errors) or there will be a specific "This version of C++ is not supported by this SYCL version". I would rather have the explicit error message, if possible.